docs/miscellaneous.rst

```diff
@@ -303,6 +303,121 @@ The following is the order of precedence for operators, listed in order of evalu
+
+``var x = uint8(258); x == 2``
+
+Table I: Rules for operators and unsigned Integers
+--------------------------------------------------
+
++------------+----------+-----------+---------+
+|  Operator  | Overflow | Underflow | Notes   |
++============+==========+===========+=========+
+| +, ++, +=  | x        |           |         |
++------------+----------+-----------+---------+
+| \+ (unary) | N/A      | N/A       | 1       |
++------------+----------+-----------+---------+
+| -, --, -=  |          | x         |         |
++------------+----------+-----------+---------+
+| \- (unary) |          |           | 2       |
++------------+----------+-----------+---------+
+| \*, \*=    | x        |           |         |
++------------+----------+-----------+---------+
+| \*\*       | x        |           | 3       |
++------------+----------+-----------+---------+
+| /, /=      |          |           |         |
++------------+----------+-----------+---------+
+| %, %=      |          |           |         |
++------------+----------+-----------+---------+
+| <<, <<=    |          |           | 4       |
++------------+----------+-----------+---------+
+| >>, >>=    |          |           | 5       |
++------------+----------+-----------+---------+
+| &, &=      |          |           |         |
++------------+----------+-----------+---------+
+| \|, \|=    |          |           |         |
++------------+----------+-----------+---------+
+| ^, ^=      |          |           |         |
++------------+----------+-----------+---------+
+| ~, ~=      |          |           |         |
++------------+----------+-----------+---------+
+
+1) Unary ``+`` has been deprecated.
+2) Unary ``-`` also works on ``uints``, e.g. ``-uint(1) == ~uint(0)``.
+3) ``0**0 := 1``
+4) Arithmetic left shift, implemented as clamped multiplication ``n << m == n * 2**m`` (overflowing bits are removed). The bitsize of the shifted value is that of ``n``.
+5) Arithmetic right shift, implemented as division ``n >> m == n / 2**m``. The bitsize of the shifted value is that of ``n``.
+
+Table II: Rules for operators and signed integers
+-------------------------------------------------
+
++------------+----------+-----------+---------+
+|  Operator  | Overflow | Underflow | Notes   |
++============+==========+===========+=========+
+| +, ++, +=  | x        | x         |         |
++------------+----------+-----------+---------+
+| \+ (unary) | N/A      | N/A       | 1       |
++------------+----------+-----------+---------+
+| -, --, -=  | x        | x         |         |
++------------+----------+-----------+---------+
+| \- (unary) | x        | x         |         |
++------------+----------+-----------+---------+
+| \*, \*=    | x        | x         |         |
++------------+----------+-----------+---------+
+| \*\*       | x        | x         | 2       |
++------------+----------+-----------+---------+
+| /, /=      |          |           |         |
```

Comment:
This actually can overflow.

I'm not sure if a table is the best way to present this information. I would prefer text that also explains what overflow means in such a case, what actually happens and why some operators cannot overflow.

Reply:
When two signed 2's complement numbers are added, overflow is detected if:

- both operands are positive and the result is negative, or

- both operands are negative and the result is positive.

When two unsigned numbers are added, overflow occurs if

- there is a carry out of the leftmost bit.


Is that good?

Reply:
also, my table has the updated rules from having edited. this is an old pr i will remove it since it has been updated so much and is getting messy.